### PR TITLE
New controls review

### PR DESCRIPTION
**Pull request by daniel-wer**
### URL of deployed dev instance (used for testing):
- https://___.webknossos.xyz

### Steps to test:
- abc

### Issues:
- fixes #

------
- [ ] Updated [(unreleased) changelog](../blob/master/CHANGELOG.unreleased.md#unreleased)
- [ ] Updated [(unreleased) migration guide](../blob/master/MIGRATIONS.unreleased.md#unreleased) if applicable
- [ ] Updated [documentation](../blob/master/docs) if applicable
- [ ] Adapted [wk-connect](https://github.com/scalableminds/webknossos-connect) if datastore API changes
- [ ] Needs datastore update after deployment
- [x] Ready for review

### CODE REVIEW

File: frontend/javascripts/oxalis/view/action-bar/toolbar_view.js [hunk 1]
```diff
@@ -330,8 +348,34 @@ export default function ToolbarView() {
                   opacity: disabledInfosForTools[AnnotationToolEnum.TRACE].isDisabled ? 0.5 : 1,
                 }}
               />
-              {adaptedActiveTool === "TRACE" ? multiSliceAnnotationInfoIcon : null}
+              {adaptedActiveTool === AnnotationToolEnum.TRACE ? multiSliceAnnotationInfoIcon : null}
+            </RadioButtonWithTooltip>
+
+            <RadioButtonWithTooltip
+              title="Erase – Draw outlines around the voxel you would like to erase."
```
Comment (daniel-wer) [inline frontend/javascripts/oxalis/view/action-bar/toolbar_view.js:355]:
```suggestion
              title="Erase – Draw outlines around the voxels you would like to erase."
```

File: frontend/javascripts/oxalis/view/action-bar/toolbar_view.js [hunk 2]
```diff
@@ -314,8 +306,34 @@ export default function ToolbarView() {
                   opacity: disabledInfosForTools[AnnotationToolEnum.BRUSH].isDisabled ? 0.5 : 1,
                 }}
               />
-              {adaptedActiveTool === "BRUSH" ? multiSliceAnnotationInfoIcon : null}
+              {adaptedActiveTool === AnnotationToolEnum.BRUSH ? multiSliceAnnotationInfoIcon : null}
+            </RadioButtonWithTooltip>
+
+            <RadioButtonWithTooltip
+              title="Erase (via Brush) – Erase the voxel by brushing over them. Adjust the brush size with Shift + Mousewheel."
```
Comment (daniel-wer) [inline frontend/javascripts/oxalis/view/action-bar/toolbar_view.js:313]:
```suggestion
              title="Erase (via Brush) – Erase voxels by brushing over them. Adjust the brush size with Shift + Mousewheel."
```

File: frontend/javascripts/oxalis/view/context_menu.js [hunk 3]
```diff
@@ -177,37 +191,65 @@ function NodeContextMenuOptions({
...
+      Select Cell ({cellIdAtPosition})
+    </Menu.Item>,
+
+    <Menu.Item
+      className="node-context-menu-item"
+      key="fill-cell"
+      onClick={() => handleFloodFillFromGlobalPosition(globalPosition, viewport)}
+    >
+      Fill Cell (flood-fill region)
+    </Menu.Item>,
```
Comment (daniel-wer) [inline frontend/javascripts/oxalis/view/context_menu.js:235]: Shouldn't these options be restricted to annotations with a segmentation layer? Or are they, and I don't see it?

File: frontend/javascripts/oxalis/view/settings/user_settings_view.js [hunk 4]
```diff
@@ -347,11 +347,6 @@ class UserSettingsView extends PureComponent<UserSettingsViewProps> {
             value={this.props.isMergerModeEnabled}
             onChange={this.props.onChangeEnableMergerMode}
             disabled={!isMergerModeSupported || this.props.isMergerModeTask}
-            tooltipText={
```
Comment (daniel-wer) [inline frontend/javascripts/oxalis/view/settings/user_settings_view.js:350]: Did you move this somewhere else or was it superfluous?

**Review by daniel-wer — COMMENTED** [review]
Tremendous work! I think you did a great job implementing the legacy mode!

Sure, there is some code that needs to deal with the two modes, but thanks to your refactorings, very few files need to know about those. Keeping in mind the complexity of wK and its keyboard and mouse handling functionality, this is solved very elegantly :+1: 

Considering the magnitude of these changes (and keeping in mind that we don't want to change the controls again in the near future) I would rather do a couple more testing and feedback rounds with users from our company, but ideally also real-world wK users.

Excited for the finishing touches, I'll review those in the original PR, then :)

**Comment by daniel-wer** [comment]
@philippotto Not sure whether you got notified since I only assigned you after reviewing. Review is done :)

File: frontend/javascripts/oxalis/controller/combinations/tool_controls.js [hunk 5]
```diff
@@ -206,39 +301,130 @@ export class DrawTool {
...
+        );
       },
 
       out: () => {
         Store.dispatch(hideBrushAction());
       },
     };
   }
+
+  static getActionDescriptors(
```
[the remark this answers is not in the document]
Comment (philippotto) [reply frontend/javascripts/oxalis/controller/combinations/tool_controls.js:359]: Good point! I extended the comment block at the top of the module.

File: frontend/javascripts/oxalis/controller/combinations/tool_controls.js [hunk 6]
```diff
@@ -206,39 +301,130 @@ export class DrawTool {
...
+      leftDownMove: (delta: Point2, pos: Point2) => {
+        VolumeHandlers.handleDrawDeleteMove(pos);
+      },
+
+      leftMouseDown: (pos: Point2, plane: OrthoView, _event: MouseEvent) => {
+        VolumeHandlers.handleEraseStart(pos, plane);
+      },
+
+      leftMouseUp: () => {
+        VolumeHandlers.handleDrawEraseEnd();
```
[the remark this answers is not in the document]
Comment (philippotto) [reply frontend/javascripts/oxalis/controller/combinations/tool_controls.js:396]: I cleaned it up a bit now :+1: Now, there's:
- handleDrawStart
- handleEraseStart
- handleMoveForDrawOrErase
- handleEndForDrawOrErase

File: frontend/javascripts/oxalis/view/context_menu.js (hunk 3 above)
Comment (philippotto) [reply frontend/javascripts/oxalis/view/context_menu.js:235]: You're right! Fixed it in the meantime :)

**Comment by philippotto** [comment]
@daniel-wer Thanks for your feedback! I think, I integrated everything. Feel free to have a look at the latest commits in https://github.com/scalableminds/webknossos/pull/5422 where I pushed these changes

**Comment by philippotto** [comment]
Closing this since this PR was only relevant for reviewing parts of #5422.

**Pull request closed without merging**